### ISSUE TITLE
[Bug]: NodeViewRendererProps typed incorrectly? 

### ISSUE THREAD

**Issue by ColemanDunn** [ueberdosis/tiptap#5483]
### Affected Packages

core

### Version(s)

2.6.2

### Bug Description

According to the [docs](https://tiptap.dev/docs/editor/extensions/custom-extensions/node-views/react#all-available-props) and the props object when you log it in the console actually exactly matches the `NodeViewProps` type. 

For example `updateAttributes` does not exist on `NodeViewRendererProps` but the docs state it is available and it is indeed usable when working with the code. Another is `HTMLAttributes` is part of `NodeViewRendererProps` but when you access it, it is undefined.

here are the types from the source code 
```ts
export type NodeViewProps = {
  editor: Editor;
  node: ProseMirrorNode;
  decorations: DecorationWithType[];
  selected: boolean;
  extension: Node;
  getPos: () => number;
  updateAttributes: (attributes: Record<string, any>) => void;
  deleteNode: () => void;
};

export interface NodeViewRendererOptions {
  stopEvent: ((props: { event: Event }) => boolean) | null;
  ignoreMutation:
    | ((props: { mutation: MutationRecord | { type: 'selection'; target: Element } }) => boolean)
    | null;
  contentDOMElementTag: string;
}

export type NodeViewRendererProps = {
  editor: Editor;
  node: ProseMirrorNode;
  getPos: (() => number) | boolean;
  HTMLAttributes: Record<string, any>;
  decorations: Decoration[];
  extension: Node;
};

export type NodeViewRenderer = (props: NodeViewRendererProps) => NodeView | {};```

I am not sure if this is just a matter of replacing the `NodeViewRendererProps` type with `NodeViewProps` but the props object does not represent that type when used with `NodeViewRenderer`.

### Browser Used

Chrome

### Code Example URL

_No response_

### Expected Behavior

I can call `updateAttributes` on the props object when using `NodeViewRenderer` without getting a TS error. 

### Additional Context (Optional)

_No response_

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by nperez0111** [comment]
I agree there is definitely mis typings here (probably because it was originally typed before prosemirror had those features and even possibly before prosemirror had TS.

I would be happy to accept a PR for this @ColemanDunn

**Comment by nperez0111** [comment]
Would also love to resolve https://github.com/ueberdosis/tiptap/issues/3770#issuecomment-2287042520 along the way

**Comment by ColemanDunn** [comment]
> Would also love to resolve [#3770 (comment)](https://github.com/ueberdosis/tiptap/issues/3770#issuecomment-2287042520) along the way

Sounds good! Although I'm not too sure on that issue above since `props.innerDecorations` is undefined on my end... Unless the suggested type in that issue is supposed to be optional for `innerDecorations: DecorationSource` I'm not sure what to do there and I am not familiar with `innerDecorations` at all.

**Comment by ColemanDunn** [comment]
> I agree there is definitely mis typings here (probably because it was originally typed before prosemirror had those features and even possibly before prosemirror had TS.
> 
> I would be happy to accept a PR for this @ColemanDunn

Hey I got some stuff pushed up! Any recommendations? https://github.com/ueberdosis/tiptap/pull/5487#issuecomment-2287690855

**Comment by nperez0111** [comment]
Hey sorry, got lost in a bunch of other PRs I'll review now

**Comment by nperez0111** [comment]
This is released with v2.7.0 https://github.com/ueberdosis/tiptap/releases/tag/v2.7.0
